src/components/TodoItem.tsx

```diff
@@ -0,0 +1,64 @@
+import cn from 'classnames';
+import { useState } from 'react';
+import { useTodoContext } from '../context/TodoContext';
+import { deleteTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { TodoError } from '../types/Error';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { setTodos, setError } = useTodoContext();
+  const [isTodoDeleting, setIsTodoDeleting] = useState(false);
+
+  const handleTodoDelete = async () => {
+    setIsTodoDeleting(true);
+
+    try {
+      await deleteTodo(todo.id);
+
+      setTodos((prevTodos) => prevTodos.filter((t) => t.id !== todo.id));
+      setIsTodoDeleting(false);
```

Comment:
you can do it in one time in `finally` block instead here and same thing in catch block